### ISSUE TITLE
Add onSubmitted to the SearchAnchor.bar widget

### ISSUE THREAD

**Issue by PsyOhm23** [flutter/flutter#130687]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Like for the recently closed issue #126551 by the PR #129365 , it will be great if we could have the `onSubmitted` handler of `SearchBar` on the factory `SearchAnchor.bar` widget.

### Proposal

Add the `onSubmitted` handler like it was done for the `SearchBar` on the `SearchAnchor.bar` widget

**Comment by PackRuble** [comment]
Any updates?

**Comment by huycozy** [comment]
@QuncCccccc There is a similar request at https://github.com/flutter/flutter/issues/136790 but the OP proposed `SearchAnchor`. I think it can be centralized on this issue, am I correct?

**Comment by SpencerRiddering** [comment]
This ticket is incomplete in that it only addresses the missing `onSubmitted` property on the `SearchAnchor.bar` factory. 

To understand the issue we must first recognize that `SearchAnchor` has two distinct experiences: "search bar" and "search view". The "search bar" (defined by a `SearchBar` instance) is returned by `SearchAnchor`'s `builder` property (and created internally by the `SearchAnchor.bar` factory). The "search view" is constructed within `SearchAnchor` when `controller.openView()` is called. 

We need to receive callbacks when the search is submitted, both from the "search bar" and from the "search view".  So, in addition to the original proposal here to add an `onSubmitted` property to the `SearchAnchor.bar` factory, we also need a `viewOnSubmitted` property added to both `SearchAnchor` and `SearchAnchor`.bar. This new `viewOnSubmitted` property would only apply to the "search view" experience. 

So in summary we need: 

`SearchAnchor.bar( onSubmitted: ..., viewOnSubmitted: ..., ...) `
`SearchAnchor( viewOnSubmitted: ..., ...) `

**Comment by QuncCccccc** [comment]
> This ticket is incomplete in that it only addresses the missing `onSubmitted` property on the `SearchAnchor.bar` factory.
> 
> To understand the full issue, we must first recognize that `SearchAnchor` has two distinct experiences: "search bar" and "search view". The "search bar" (defined by a `SearchBar` instance) is returned by `SearchAnchor`'s `builder` property (and created internally by the `SearchAnchor.bar` factory). The "search view" is constructed within `SearchAnchor` when `controller.openView()` is called.
> 
> We need to receive callbacks when the search is submitted, both from the "search bar" and from the "search view". So, in addition to the original proposal here to add an `onSubmitted` property to the `SearchAnchor.bar` factory, we also need a `viewOnSubmitted` property added to both `SearchAnchor` and `SearchAnchor.bar`. This new `viewOnSubmitted` property would only apply to the "search view" experience.
> 
> So in summary we need:
> 
> `SearchAnchor.bar( onSubmitted: ..., viewOnSubmitted: ..., ...) ` `SearchAnchor( viewOnSubmitted: ..., ...) `

Thanks for the explanation! I'm going to work on this very soon:)

**Comment by SpencerRiddering** [comment]
> > This ticket is incomplete in that it only addresses the missing `onSubmitted` property on the `SearchAnchor.bar` factory.
> > To understand the full issue, we must first recognize that `SearchAnchor` has two distinct experiences: "search bar" and "search view". The "search bar" (defined by a `SearchBar` instance) is returned by `SearchAnchor`'s `builder` property (and created internally by the `SearchAnchor.bar` factory). The "search view" is constructed within `SearchAnchor` when `controller.openView()` is called.
> > We need to receive callbacks when the search is submitted, both from the "search bar" and from the "search view". So, in addition to the original proposal here to add an `onSubmitted` property to the `SearchAnchor.bar` factory, we also need a `viewOnSubmitted` property added to both `SearchAnchor` and `SearchAnchor.bar`. This new `viewOnSubmitted` property would only apply to the "search view" experience.
> > So in summary we need:
> > `SearchAnchor.bar( onSubmitted: ..., viewOnSubmitted: ..., ...) ` `SearchAnchor( viewOnSubmitted: ..., ...) `
> 
> Oh I just noticed that you also proposed `onSubmitted` for SearchAnchor.bar. Just wonder what the use case could be? I thought a new viewOnSubmitted in SearchAnchor.bar would be enough:)

@QuncCccccc That's a good question: does SearchAnchor.bar require separate onSubmit handlers for the "search bar" and "search view" experiences, or would a single property be sufficient? I think a single parameter is sufficient. I would call the SearchAnchor.bar parameter "onSubmit" because "viewOnSubmitted" would indicate (in my mind) that the parameter will only be applied to the "search view" experience.
